File: packages/gbnf-experiment/data/20260909T132655Z_2a16d72b/ported_implementation/gbnf/rules_builder/parse_space.py
def char_at(src: str, pos: int) -> str:
    """Mirror JS string indexing: out of range reads are falsy rather than an error."""
    if 0 <= pos < len(src):
        return src[pos]
    return ""


def parse_space(src: str, pos: int, newline_ok: bool) -> int:
    while char_at(src, pos) in (" ", "\t", "#") or (
        newline_ok and char_at(src, pos) in ("\r", "\n")
    ):
        if char_at(src, pos) == "#":
            while char_at(src, pos) and char_at(src, pos) not in ("\r", "\n"):
                pos += 1
        else:
            pos += 1
    return pos

File: proposed-projects/round-trip-experiment/reverse/20260909T133702Z_8305aa82/20260910T075701Z_f285a71f/ported_implementation/src/rules-builder/parse-space.ts
export const parseSpace = (
  src: string,
  pos: number,
  newlineOk: boolean,
): number => {
  const at = (i: number): string => (i >= 0 && i < src.length ? src[i] : '');

  while (
    at(pos) === ' ' ||
    at(pos) === '\t' ||
    at(pos) === '#' ||
    (newlineOk && (at(pos) === '\r' || at(pos) === '\n'))
  ) {
    if (at(pos) === '#') {
      while (at(pos) && at(pos) !== '\r' && at(pos) !== '\n') {
        pos += 1;
      }
    } else {
      pos += 1;
    }
  }
  return pos;
};
